src/System.Private.Xml/src/System/Xml/Serialization/PrimitiveXmlSerializers.cs

```diff
@@ -236,11 +236,8 @@ protected override void InitCallbacks()
         }
     }
 
-#if XMLSERIALIZERGENERATOR
+#if !XMLSERIALIZERGENERATOR
     internal class XmlSerializationPrimitiveReader : Microsoft.XmlSerializer.Generator.XmlSerializationReader
```

Comment:
I had planned to exclude the XmlSerializationPrimitiveReader class in sgen but end with the wrong code here. Will update soon.